### ISSUE TITLE
Request for other distributions ?

### ISSUE THREAD

**Issue by localredhead** [reynhout/chrx#13]
Thank you so much for this project.  It is amazing and has enabled a better form of computing for myself.

I am curious if other distributions that claim to support hidpi or claim to be tuned for Chromebooks can also be considered?  

Of note, these are on my short list:
- https://cublinux.com/
- https://ubuntu-mate.org/

**Comment by reynhout** [comment]
I'd like to add chrx support for these and other distributions. My time is limited presently, so I can't commit to it on any sort of timeline, but I would also welcome PRs.

FWIW, my understanding is that Ubuntu MATE and Cub Linux are not actually tuned for Chromebooks. They are generally light-weight, but made for (and therefore only include drivers for) standard-issue x86 desktops and laptops. In this way they resemble Xubuntu and Lubuntu, whereas GalliumOS is the opposite.

Ubuntu MATE and Cub Linux are both based on Ubuntu though, so it should be possible to add support for them relatively easily. Most of the same processes should work mostly the same way. Ubuntu MATE is an official flavor of Ubuntu now, so would probably be straightforward to add. Cub Linux would likely take some additional effort but could reuse some of the mechanisms added for GalliumOS support.

**Comment by localredhead** [comment]
If I can make time I will see if I can help contribute.  

Now that it is super easy to install on external media using Chrx I can experiment quickly!

I am actually quite happy with Gallium - although I wish I could manage windows better using the touchscreen.. all I can do is scroll.  

All themes have buttons that are too small for me to use easily on the touch screen and that is the reason why I'd like to experiment (Easily) using Chrx - to check out how these other distro's are doing hidpi.

**Comment by pboling** [comment]
I am hoping to be able to install from an ISO file provided as a command line option as well.  I have had to reinstall many times and it doesn't seem efficient to have to download it every time.  If command line ISO support is added it should allow experimentation with other flavors from distro share or elsewhere.

**Comment by reynhout** [comment]
@pboling I don't think it's possible for chrx to install from a local ISO. See https://github.com/reynhout/chrx/issues/10 for previous discussion.

**Comment by mrjohnc** [comment]
Question, what are the barriers to offering installing Windows 10 as an option? I understand its not free software but is this the only barrier?

I know there are is another option to install windows but that requires installing a new BIOS and I'm not able to do this

Thanks

**Comment by HUSKI3** [comment]
You can install Windows 10, but its a very big pain. First of all, different Kernels. No might have written the right drivers for your parts. In some cases you might be lucky and put together drivers that magically work. But on my machine at least, to get that running I needed to use a usb keyboard and mouse which barely worked, but did. Now in the case of Chrx, as I understand its mainly for Linux based, as from what I see its a script that uses ChromeOS shell to modify a chroot which is then installed on the partition. 
TLDR: Windows is possible, but a big pain to get working, and not quite what Chrx is meant for.

**Comment by mrjohnc** [comment]
OK, thanks @HUSKI3  are there any instructions anywhere for this to try it? I'm not expecting it to work but I may as well try

**Comment by HUSKI3** [comment]
@mrjohnc That greatly depends on your Chromebook, instructions might vary for different models. But usually to start the installation it should be as easy as making a Windows bootable drive and using the boot menu provided by Mr.Chromebox in his firmware to select the drive. The rest of the installation depends on the chromebook, normally you can just run the installation and side-load the drivers using the CMD in the recovery menu, or through the Device Manager if you are lucky to get into it. I can help you further, but for that I will need your Chromebook model.

**Comment by mrjohnc** [comment]
@HUSKI3 thank you very much for the explanation and offer, as far as I understand my laptop Lenovo Yoga Chromebook C630 is not yet supported yet

**Comment by HUSKI3** [comment]
@mrjohnc Thats quite a powerfull chromebook. In theory if im not wrong, you can try tutorials for Pixel books. You should use the chrx tool to partition your disk as it has a partitioning utility, then install some light linux in the partition (You will overwrite it with the Windows installation). Then flash windows onto a usb stick. Go back to the chromebook, reboot it and press ctrl+l to enter GRUB, select the windows usb and it will start the installation. You will need a usb mouse and keyboard for this part. Select the drive you just created and install Windows onto it. Then you will need to go to the device manager and select the recongnized devices, if possible try to find drivers for keyboard and mouse. Then you should be able to use that installation.
Hope this helps!

**Comment by reynhout** [comment]
@mrjohnc I don't know if they're still active, but https://reddit.com/r/chrultrabook used to be a good resource for Windows-on-Chromebook info.

**Comment by mrjohnc** [comment]
Thanks very much both

**Comment by RedcatPhoenix2** [comment]
> I'd like to add chrx support for these and other distributions. My time is limited presently, so I can't commit to it on any sort of timeline, but I would also welcome PRs.
> 
> FWIW, my understanding is that Ubuntu MATE and Cub Linux are not actually tuned for Chromebooks. They are generally light-weight, but made for (and therefore only include drivers for) standard-issue x86 desktops and laptops. In this way they resemble Xubuntu and Lubuntu, whereas GalliumOS is the opposite.
> 
> Ubuntu MATE and Cub Linux are both based on Ubuntu though, so it should be possible to add support for them relatively easily. Most of the same processes should work mostly the same way. Ubuntu MATE is an official flavor of Ubuntu now, so would probably be straightforward to add. Cub Linux would likely take some additional effort but could reuse some of the mechanisms added for GalliumOS support.

I've used Ubuntu MATE on my Chromebook, all you need to do is install the MATE Desktop Environment thru the Ubuntu Terminal and run it!

**Comment by mrjohnc** [comment]
> > I'd like to add chrx support for these and other distributions. My time is limited presently, so I can't commit to it on any sort of timeline, but I would also welcome PRs.
> > FWIW, my understanding is that Ubuntu MATE and Cub Linux are not actually tuned for Chromebooks. They are generally light-weight, but made for (and therefore only include drivers for) standard-issue x86 desktops and laptops. In this way they resemble Xubuntu and Lubuntu, whereas GalliumOS is the opposite.
> > Ubuntu MATE and Cub Linux are both based on Ubuntu though, so it should be possible to add support for them relatively easily. Most of the same processes should work mostly the same way. Ubuntu MATE is an official flavor of Ubuntu now, so would probably be straightforward to add. Cub Linux would likely take some additional effort but could reuse some of the mechanisms added for GalliumOS support.
> 
> I've used Ubuntu MATE on my Chromebook, all you need to do is install the MATE Desktop Environment thru the Ubuntu Terminal and run it!

This is a really good point, it would be really nice to have a note on the instruction page, something like 'Note: Ubuntu Mate is available for devices with less resources, to use Ubuntu Mate install Ubuntu and then install the Mate desktop using these instructions' (with a link)

**Comment by RedcatPhoenix2** [comment]
> > > I'd like to add chrx support for these and other distributions. My time is limited presently, so I can't commit to it on any sort of timeline, but I would also welcome PRs.
> > > FWIW, my understanding is that Ubuntu MATE and Cub Linux are not actually tuned for Chromebooks. They are generally light-weight, but made for (and therefore only include drivers for) standard-issue x86 desktops and laptops. In this way they resemble Xubuntu and Lubuntu, whereas GalliumOS is the opposite.
> > > Ubuntu MATE and Cub Linux are both based on Ubuntu though, so it should be possible to add support for them relatively easily. Most of the same processes should work mostly the same way. Ubuntu MATE is an official flavor of Ubuntu now, so would probably be straightforward to add. Cub Linux would likely take some additional effort but could reuse some of the mechanisms added for GalliumOS support.
> > 
> > 
> > I've used Ubuntu MATE on my Chromebook, all you need to do is install the MATE Desktop Environment thru the Ubuntu Terminal and run it!
> 
> This is a really good point, it would be really nice to have a note on the instruction page, something like 'Note: Ubuntu Mate is available for devices with less resources, to use Ubuntu Mate install Ubuntu and then install the Mate desktop using these instructions' (with a link)

All you need to do is `sudo apt install [desktop-environment]` to install XFCE, LXDE, KDE, MATE, Unity, and more!
